src/codegeneration/ES6ClassTransformer.js

```diff
@@ -0,0 +1,259 @@
+import {TempVarTransformer} from './TempVarTransformer.js';
+import {ParseTreeTransformer} from './ParseTreeTransformer.js';
+import {
+  createBindingIdentifier,
+  createCommaExpression,
+  createEmptyParameterList,
+  createExpressionStatement,
+  createFunctionBody,
+  createIdentifierToken,
+  createImmediatelyInvokedFunctionExpression,
+  createPropertyNameAssignment,
+  createParenExpression,
+  createThisExpression,
+} from './ParseTreeFactory.js';
+import {
+  parseExpression,
+  parsePropertyDefinition,
+  parseStatement,
+} from './PlaceholderParser.js';
+import {propName} from '../staticsemantics/PropName.js';
+import {prependStatements} from './PrependStatements.js';
+
+/**
+ * Transform member variable declarations to valid ES6 statements.
+ *
+ * - instance variables are initialized in the constructor,
+ * - static variables are initialized after the class definition,
+ *   through ` Object.defineProperty(...)`
+ */
+export class ES6ClassTransformer extends TempVarTransformer {
+  transformClassElements_(tree) {
+    let elements = [];
+    let initInstanceVars = [], initStaticVars = [];
+    let constructor;
+    let constructorIndex = 0;
+
+    tree.elements.forEach((tree) => {
+      let initVars;
+      if (tree.isStatic) {
+        initVars = initStaticVars;
+      } else {
+        initVars = initInstanceVars;
+      }
+
+      switch (tree.type) {
+        case GET_ACCESSOR:
+        case SET_ACCESSOR:
+          elements.push(this.transformAny(tree));
+          break;
+
+        case PROPERTY_METHOD_ASSIGNMENT:
+          if (!tree.isStatic && propName(tree) === CONSTRUCTOR) {
+            constructor = tree;
+            constructorIndex = elements.length;
+          } else {
+            elements.push(this.transformAny(tree));
+          }
+          break;
+
+        case PROPERTY_VARIABLE_DECLARATION:
+          tree = this.transformAny(tree);
+          if (tree.initializer !== null) {
+            initVars.push(tree);
+          }
+          break;
+
+        default:
+          throw new Error(`Unexpected class element: ${tree.type}`);
+      }
+    });
+
+    if (initInstanceVars.length > 0) {
+      let initExpression = getInstanceInitExpression(initInstanceVars);
+
+      if (!constructor) {
+        constructor = this.getDefaultConstructor_(!!tree.superClass);
+      }
+
+      constructor = this.appendInstanceInitializers_(constructor,
+          initExpression, tree.superClass);
+    }
+
+    if (constructor) {
+      elements.splice(constructorIndex, 0, constructor);
+    }
+
+    return {
+      elements,
+      initStaticVars,
+    };
+  }
+
+  /**
+   * Transforms a single class declaration
+   *
+   * @param {ClassDeclaration} tree
+   * @return {ParseTree}
+   */
+  transformClassDeclaration(tree) {
+    let {
+      elements,
+      initStaticVars,
+    } = this.transformClassElements_(tree);
+
+    let statements = createStaticInitializerStatements(tree.name, initStaticVars);
```

Comment:
For the ClassExpression you checked the length of initStaticVars. Maybe do the same here.